Fast Pair locator tag: release configuration (#15533)
nrfconnect/sdk-nrf: samples/bluetooth/fast_pair/locator_tag/README.rst

   You cannot use the sample without the Fast Pair provisioning data.
   Programming the device with the sample firmware without providing the proper Fast Pair provisioning data results in assertion failure during boot.

Release build
=============

> You can build the sample with a release configuration to demonstrate how to adapt the default sample configuration for production, using the ``-DFILE_SUFFIX=release`` flag in your build.

Reviewer: Should we state `to adapt the default sample configuration for production`? Maybe we should only inform that it disables assertions+logs, enables LTO and it optimizes memory consumption? Keep in mind that we do not even support bootloader here, so it's still a sample - not reference design

> You can build the sample with a release configuration as a demonstration of how to adapt the default sample configuration for production, using the ``-DFILE_SUFFIX=release`` flag in your build.

Reviewer 2: ```suggestion
You can build the sample with a release configuration to demonstrate how to adapt the default sample configuration for production, using the ``-DFILE_SUFFIX=release`` flag in your build.
```

> The release build reduces code size and RAM usage of the sample by disabling the logging functionality and by performing other optimizations.
> The release build also enables the Link Time Optimization (LTO) configuration - :kconfig:option:`CONFIG_LTO` - to further reduce the code size.
> LTO is an advanced compilation technique that performs optimization across all compiled units of an application at the link stage, rather than within each unit separately.

Reviewer 2: ```suggestion
The release build reduces the code size and RAM usage of the sample by disabling logging functionality and performing other optimizations. 
Additionally, it enables the Link Time Optimization (LTO) configuration through the :kconfig:option:`CONFIG_LTO` Kconfig option, which further reduces the code size. 
LTO is an advanced compilation technique that optimizes across all compiled units of an application at the link stage, rather than optimizing each unit separately.
```

.. note::

>    Support for LTO is experimental.

Reviewer 2: ```suggestion
   Support for the LTO is experimental.
```

> See :ref:`cmake_options` for instructions on how to add this option to your build.
> For example, when building on the command line, you can add the option as follows:

Reviewer 2: ```suggestion
See :ref:`cmake_options` for detailed instructions on how to add this option to your build.
For example, when building from the command line, you can add it as follows:
```

   You cannot use the sample without the Fast Pair provisioning data.
   Programming the device with the sample firmware without providing the proper Fast Pair provisioning data results in assertion failure during boot.

Release build
=============

> You can build the sample in a release variant using the ``-DFILE_SUFFIX=release`` flag in your build command to learn about the configuration changes that you should apply to prepare the production build of your end product.

Reviewer: ```suggestion
You can build the sample in a release variant setting the ``FILE_SUFFIX=release`` CMake option.
That would result in using :file:`prj_release.conf` configuration file instead of :file:`prj.conf` configuration file for the build.
Check content of both files to learn about the configuration changes that you should apply to prepare the production build of your end product.
```
I would inform that user should check file content for details here.

Reviewer: Also we should link to doc section about setting CMake options here (keep in mind that many customers do not build from CLI)

   You cannot use the sample without the Fast Pair provisioning data.
   Programming the device with the sample firmware without providing the proper Fast Pair provisioning data results in assertion failure during boot.

Release build
=============

> You can build the sample in a release variant setting the ``FILE_SUFFIX=release`` CMake option.

Reviewer 2: ```suggestion
To build the sample in a release variant, set the ``FILE_SUFFIX=release`` CMake option.
```

> That would result in using :file:`prj_release.conf` configuration file instead of :file:`prj.conf` configuration file for the build.

Reviewer 2: ```suggestion
The build will use the :file:`prj_release.conf` configuration file instead of :file:`prj.conf`.
```

> Check content of both files to learn about the configuration changes that you should apply to prepare the production build of your end product.

Reviewer 2: ```suggestion
Check the contents of both files to learn which configuration changes you should apply when preparing the production build of your end product.
```
